src/Ng/Jobs/MonitoringProcessorJob.cs

```diff
@@ -0,0 +1,178 @@
+        private IStorageQueue<PackageValidatorContext> _queue;
+        private IPackageMonitoringStatusService _statusService;
+        private IMonitoringNotificationService _notificationService;
+        private RegistrationResourceV3 _regResource;
+        private NuGet.Common.ILogger CommonLogger;
+        private CollectorHttpClient _client;
+
+        public MonitoringProcessorJob(ILoggerFactory loggerFactory)
+            : base(loggerFactory)
+        {
+            CommonLogger = Logger.AsCommon();
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            var gallery = arguments.GetOrThrow<string>(Arguments.Gallery);
+            var index = arguments.GetOrThrow<string>(Arguments.Index);
+            var source = arguments.GetOrThrow<string>(Arguments.Source);
+            var verbose = arguments.GetOrDefault(Arguments.Verbose, false);
+
+            if (arguments.GetOrThrow<string>(Arguments.StorageType) != Arguments.AzureStorageType)
+            {
+                throw new ArgumentException("File storage is not supported!");
+            }
+
+            var monitoringStorageFactory = CommandHelpers.CreateStorageFactory(arguments, verbose);
+            var auditingStorageFactory = CommandHelpers.CreateSuffixedStorageFactory("Auditing", arguments, verbose);
+
+            var endpointInputs = CommandHelpers.GetEndpointFactoryInputs(arguments);
+
+            var messageHandlerFactory = CommandHelpers.GetHttpMessageHandlerFactory(verbose);
+
+            Logger.LogInformation(
+                "CONFIG gallery: {Gallery} index: {Index} storage: {Storage} auditingStorage: {AuditingStorage} endpoints: {Endpoints}",
+                gallery, index, monitoringStorageFactory, auditingStorageFactory, string.Join(", ", endpointInputs.Select(e => e.Name)));
+
+            _packageValidator = new PackageValidatorFactory(LoggerFactory)
+                .Create(gallery, index, auditingStorageFactory, endpointInputs, messageHandlerFactory, verbose);
+
+            _queue = CommandHelpers.CreateStorageQueue<PackageValidatorContext>(arguments);
+
+            _statusService = CommandHelpers.GetPackageMonitoringStatusService(arguments, monitoringStorageFactory, LoggerFactory);
+
+            _notificationService = new LoggerMonitoringNotificationService(LoggerFactory.CreateLogger<LoggerMonitoringNotificationService>());
+
+            _regResource = Repository.Factory.GetCoreV3(index).GetResource<RegistrationResourceV3>(cancellationToken);
+
+            _client = new CollectorHttpClient(messageHandlerFactory());
+        }
+
+        protected override async Task RunInternal(CancellationToken cancellationToken)
+        {
+            await ParallelExecutionUtility.ProcessInParallel(() => ProcessPackages(cancellationToken));
+        }
+
+        private async Task ProcessPackages(CancellationToken token)
```

Comment:
This method is quite large. Please move pieces out into helpers where possible

Reply:
done